bundles/org.eclipse.swt/Eclipse SWT Browser/win32/org/eclipse/swt/browser/Edge.java

```diff
@@ -282,6 +279,95 @@ static int callAndWait(String[] pstr, ToIntFunction<IUnknown> callable) {
 	return phr[0];
 }
 
+class WebViewProvider {
+
+	private CompletableFuture<ICoreWebView2> webViewFuture = new CompletableFuture<>();
+	private CompletableFuture<ICoreWebView2_2> webView_2Future = new CompletableFuture<>();
+	private CompletableFuture<ICoreWebView2_11> webView_11Future = new CompletableFuture<>();
+
+	private CompletableFuture<Void> lastWebViewTask = webViewFuture.thenCompose(__ -> {
+		return CompletableFuture.allOf(webView_2Future.handle((result, ex) -> null), webView_11Future.handle((result, ex) -> null));
+	});
+
+
+	private ICoreWebView2 getWebView(boolean waitForPendingWebviewTasksToFinish) {
+		if(waitForPendingWebviewTasksToFinish) {
+			waitForPendingWebviewTasksToFinish();
+		}
+		return webViewFuture.join();
+	}
+
+	private void initializeWebView(ICoreWebView2 webView) {
+		webViewFuture.complete(webView);
+	}
+
+	private boolean isWebViewUnavailable() {
+		return webViewFuture.isCancelled();
+	}
+
+	private ICoreWebView2_2 getWebView_2(boolean waitForPendingWebviewTasksToFinish) {
+		if(waitForPendingWebviewTasksToFinish) {
+			waitForPendingWebviewTasksToFinish();
+		}
+		return webView_2Future.join();
+	}
+
+	private void initializeWebView_2(ICoreWebView2_2 webView_2) {
+		webView_2Future.complete(webView_2);
+	}
+
+	private void cancelWebView_2Initialization() {
+		webView_2Future.cancel(true);
+	}
+
+	private boolean isWebView_2Unavailable() {
+		return webView_2Future.isCancelled();
+	}
+
+	private ICoreWebView2_11 getWebView_11(boolean waitForPendingWebviewTasksToFinish) {
+		if(waitForPendingWebviewTasksToFinish) {
+			waitForPendingWebviewTasksToFinish();
+		}
+		return webView_11Future.join();
+	}
+
+	private void initializeWebView_11(ICoreWebView2_11 webView_11) {
+		webView_11Future.complete(webView_11);
+	}
+
+	private void cancelWebView_11Initialization() {
+		webView_11Future.cancel(true);
+	}
+
+	private boolean isWebView_11Unavailable() {
+		return webView_11Future.isCancelled();
+	}
+
+	/*
+	 * Schedule a given consumer with the given webViewFuture in the webViewTask
+	 * queue (using lastWebViewTask).
+	 * Note: This will be executed in the thread from
+	 * which it is called and must be therefore called from the UI thread.
+	 */
+	private void scheduleWebViewTask(Consumer<Object> action) {
+		lastWebViewTask = lastWebViewTask.thenApply(__ -> {
+			action.accept(null);
+			// Sometimes when the shell of the browser is opened before the browser is
+			// initialized, nothing is drawn on the shell. We need browserResize to force
+			// the shell to draw itself again.
+			browserResize(new Event());
```

Comment:
Is it necessary to do this in every schuled task? Isn't is suffient to do it once after initialization of the webview has finieshed?